### ISSUE TITLE
[Feature Request] Save loop mode: sequential/shuffle

### ISSUE THREAD

**Issue by yunyuyuan** [jmshrv/finamp#818]
I'm using the beta version `0.9.8`.

Currently, the default loop mode is set to sequential play. Users can manually switch to shuffle play, but this setting is not saved and must be set each time. This is inconvenient for users who prefer a shuffle loop mode.

I would like finamp to save the loop mode (sequential, shuffle) for each playlist. This way, users don't have to manually switch to their preferred loop mode every time they start play a list. 

Alternatively, having a global setting to set the default loop mode would also be helpful.

Thank you for considering this feature request!

**Comment by HeroBrine1st** [comment]
Spotify has even better approach. Currently, Finamp has two buttons that do *almost* the same function: start playing playlist. Spotify, however, has toggle for shuffle and dedicated button to play (see screenshots below). This way there is only one "primary" button.

<details>
  <summary>Images</summary>
On mobile:

![image](https://github.com/user-attachments/assets/2e63ee5b-be53-4061-85bb-de9c73115693)

On PC:

![image](https://github.com/user-attachments/assets/849fee56-c0ef-4efb-b9a7-0d307d37a592)

![image](https://github.com/user-attachments/assets/cc084ada-e6d0-4f31-8f4b-c9bcfe37f9d7)

</details>

**Comment by Chaphasilor** [comment]
Yeah these are both valid suggestions. It should be fairly easy to implement too.  
Just to be sure, you want regular shuffle, but with the first track being the selected one? As opposed to starting at a certain track and then only shuffling the *remaining* tracks.

**Comment by yunyuyuan** [comment]
@Chaphasilor Yes,my current approach is to click on a track in the playlist to start playing, and then switch to shuffle mode.

**Comment by BloodyIron** [comment]
Isn't this a duplicate of #797 ?

**Comment by lukaslindnermusic** [comment]
Hmm...
This is fancy and all, but I'm honestly not sure if this makes it that much easier compared to if you just have all Playback Options directly available with just one tap or a swipe+tap, with the most recently used option presented to you by default.
And you would also not just have one play or shuffle button, but you can quickly add to queue or next up as well (which for example is my personal preference most of the time).
I think it is way more logical if each button has a clear and consistent function, while with this approach, the switch changes what the Play button does.

Besides that: We have more Playback Options than Spotify. As mentioned before, for Artists (and soon also for Genres), we have the "Shuffle Albums" function in addition to normal Play and track shuffle. How would this translate to the Shuffle-Switch Design? Would it be an extra switch?

**Comment by lukaslindnermusic** [comment]
Yeah, I think we could add a page-lock.
What do you think @Chaphasilor ?

Regarding the third category ("Shuffle Albums", not Artists):
If the page-lock is unlocked, it would just default to either Play oder Shuffle, as the Shuffle artists is a "special" playback mode.
The Playback Order of Shuffle Albums should be "normal", as it only shuffles the albums, but the queue is working on a track-basis.
As mentioned in my other comment, it should basically be the same as manually picking some albums and adding them non-shuffled to the queue.

switching the PlaybackActionRow Page should then don't change anything... only hitting play will move the default page and change the PlaybackOrder.
Unless you lock the page for the current item, then the locked page should get changed as soon as you change it on that item.

Would you like the lock to 
a) apply only to the current item, like a specific playlist or artist?
b) apply to the same type, like for example all artists simultaneously?

**Comment by HeroBrine1st** [comment]
> As mentioned in my other comment, it should basically be the same as manually picking some albums and adding them non-shuffled to the queue.

Yes, but my concern is that in manual case the shuffle mode would actually be respected.

> switching the PlaybackActionRow Page should then don't change anything... only hitting play will move the default page and change the PlaybackOrder.

As well as pressing switch should not change player state (but switch the page ofc, and vise versa). Actually we can entirely ignore the synchronisation of switch with player state, because on action row play and shuffle are visible as different buttons, so if you pressed "Play" and then switched a switch, player should not be "shuffled" because it is technically a different button.

This involves adding a new field to FinampSettings or to any other persistent Hive box, namely `boolean defaultShuffleMode`.

>  b) apply to the same type, like for example all artists simultaneously?

Absolutely against. Most of my playlists are played sequentally but some of them intended for shuffle.

A setting for separating defaults between collection types (and ofc separating from player state) can be introduced instead, but lock should still lock only one collection at a time. Then, the `defaultShuffleMode` is actually a tuple of three with a setting to synchronise them all.

**Comment by Chaphasilor** [comment]
> UI communicates that "play" action is primary, while muscular memory will direct to other buttons if needed. 

that's a good point. when switching to the `PlaybackActionsRow`, I should keep this in mind :)

I'm not yet convinced that "locking" the default order for some items but not for others is really needed. There could be a regular setting for remembering the default per-item or globally, but for now I think simply remembering it globally is a good way to get started.  
If we get multiple requests for remembering it per-item, we can still add that later.

> As well as pressing switch should not change player state (but switch the page ofc, and vise versa). Actually we can entirely ignore the synchronisation of switch with player state, because on action row play and shuffle are visible as different buttons, so if you pressed "Play" and then switched a switch, player should not be "shuffled" because it is technically a different button.

There should be no two-way sync here. Player oder is still set when starting playback via `replaceWholeQueue()`, and if the shuffle state is changed we *could* update a separate setting that remembers the current "preferred" order. And this preferred order is used for selecting the default page on the `PlaybackActionsRow`, and is updated when starting playback via the `PlaybackActionsRow`.  
Player order should *never* be changed as a side effect.

I think we've answered all questions now, just use the `PlaybackActionsRow` and remember the "preferred" page when playback is started through it.

**Comment by lukaslindnermusic** [comment]
If that's okay for everyone, I will open up a PR in a bit 😊 
Switching the play buttons to the PlaybackActionRow is something I actually already wanted to do in my last PR from a few days ago, so this would align with that. 😊
